components/locale_canonicalizer/src/locale_canonicalizer.rs

```diff
@@ -62,91 +62,94 @@ impl LocaleCanonicalizer<'_> {
     /// assert_eq!(locale.to_string(), "en-Latn-DE");
     /// # } // feature = "provider_serde"
     /// ```
-    pub fn maximize(&self, locale: &mut Locale) -> CanonicalizationResult {
-        let update_locale =
-            |entry: &LanguageIdentifier, locale: &mut Locale| -> CanonicalizationResult {
-                if locale.language.is_empty() {
-                    locale.language = entry.language;
+    pub fn maximize<T: AsMut<LanguageIdentifier>>(&self, mut langid: T) -> CanonicalizationResult {
+        let update_langid = |entry: &LanguageIdentifier,
+                             langid: &mut LanguageIdentifier|
+         -> CanonicalizationResult {
+            if langid.language.is_empty() {
+                langid.language = entry.language;
+            }
+            langid.script = langid.script.or(entry.script);
+            langid.region = langid.region.or(entry.region);
+            CanonicalizationResult::Modified
+        };
+
+        |langid: &mut LanguageIdentifier| -> CanonicalizationResult {
```

Comment:
Hmm, I haven't tested it but wouldn't that also work:

```rust
let mut langid = langid.as_mut();
// go on with mutable reference?
```

instead of this closure?

Reply:
Yes, you're right. I was trying variations of things like:
```
let &langid = &langid.as_mut();
```
yesterday and getting nowhere with it, it complains about moving out of the reference, but I somehow missed trying the most straightforward approach :/.